.github/workflows/pre-submit.actions.yml

```diff
@@ -115,6 +115,50 @@ jobs:
           SLSA_TOKEN: ${{ steps.setup.outputs.slsa-token }}
         run: ./.github/workflows/scripts/pre-submit.actions/setup-token.sh
 
+  verify-token:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
+      - id: token
+        run: |
+          set -euo pipefail
+          # Note: the token is always invalid due to the GitHub run verification.
+          token=$(cat .github/actions/verify-token/testdata/slsa-token)
+          verified_token=$(echo "$token" | cut -d '.' -f2 | base64 -d)
+          echo "unverified=$token" >> "$GITHUB_OUTPUT"
+          echo "verified=$verified_token" >> "$GITHUB_OUTPUT"
+      # Verify the values in the "verified" token.
+      - run: ./.github/workflows/scripts/schedule.actions/verify-token.sh
+        env:
+          VERIFIED_TOKEN: ${{ steps.token.outputs.verified }}
+          TOOL_REPOSITORY: "laurentsimon/slsa-delegated-tool"
+          TOOL_REF: "refs/heads/main"
+          TOOL_URI: "https://github.com/laurentsimon/slsa-delegated-tool/.github/workflows/tool1_slsa3.yml@refs/heads/main"
+      - id: verify
+        uses: ./.github/actions/verify-token
+        continue-on-error: true
+        with:
+          # Always fails due to the GitHub run mismatch.
+          slsa-unverified-token: ${{ steps.token.outputs.unverified }}
+          slsa-workflow-recipient: "delegator_generic_slsa3.yml"
+      - id: verify-mismatch-recipient
+        uses: ./.github/actions/verify-token
+        continue-on-error: true
+        with:
+          slsa-unverified-token: ${{ steps.token.outputs.unverified }}
+          slsa-workflow-recipient: "elegator_generic_slsa3.yml"
+      - id: verify-mismatch-token
+        uses: ./.github/actions/verify-token
+        continue-on-error: true
+        with:
+          slsa-unverified-token: aGVsbG8K
+          slsa-workflow-recipient: "delegator_generic_slsa3.yml"
+      - env:
+          SUCCESS: ${{ steps.verify.outcome == 'failure' && steps.verify-mismatch-recipient.outcome == 'failure' && steps.verify-mismatch-token.outcome == 'failure' }}
```

Comment:
These are all failure scenarios. Maybe add a successful test with the output of the `setup-token` test above?

Reply:
cannot use setup-token in pre-submit, because it does not have the permission for signing (`id-token: write`). The end-to-end test is done in the scheduled workflow run [(already present in the codebase)](https://github.com/slsa-framework/slsa-github-generator/blob/main/.github/workflows/e2e.verify-token.schedule.yml)